sys/console/minimal/src/console.c

```diff
@@ -70,16 +71,73 @@ console_echo(int on)
     echo = on;
 }
 
+int
+console_lock(int timeout)
+{
+    int rc = OS_OK;
+
+    /* Locking from isr while some task own mutex fails with OS_EBUSY */
+    if (os_arch_in_isr()) {
+        if (os_mutex_get_level(&console_write_lock)) {
+            rc = OS_EBUSY;
+        }
+        goto end;
+    }
+
+    rc = os_mutex_pend(&console_write_lock, timeout);
+    if (rc == OS_NOT_STARTED) {
+        /* No need to block before system start make it OK */
+        rc = OS_OK;
+    }
+
+    end:
```

Comment:
Nitpick: for consistency, this `end:` label should not be indented.

Reply:
done